pysmiles/smiles_helper.py

```diff
@@ -448,59 +439,152 @@ def _prune_nodes(nodes, mol):
             new_nodes.append(node)
     return mol.subgraph(new_nodes)
 
-def mark_aromatic_atoms(mol, atoms=None):
+
+def mark_aromatic_atoms(mol, strict=True):
     """
-    Properly kekeulizes molecules and sets the aromatic attribute.
+    Aromaticity is defined here as regions that show delocalization induced
+    molecular equivalence (DIME). In other words, regions where alternating
+    single and double bonds can interchange without inducing any charges.
+    This function will correctly identify and mark aromatic atoms within
+    ``mol``, as well as kekulize regions that are marked as aromatic, but do not
+    show DIME.
 
     Parameters
     ----------
     mol : nx.Graph
         The molecule.
-    atoms: collections.abc.Iterable
-        The atoms to act on; all other nodes are pruned
+    strict : bool
+        Whether to raise a SyntaxError if the aromatic region of the molecule
+        cannot be kekulized.
 
     Returns
     -------
     None
         `mol` is modified in-place.
+
+    Raises
+    ------
+    SyntaxError
+        If ``strict`` is True and the aromatic region of the molecule cannot
+        be kekulized.
     """
-    if atoms is None:
-        atoms = set(mol.nodes)
     # prune all nodes from molecule that are eligible and have
     # full valency
-    ds_graph = _prune_nodes(atoms, mol)
+    # ALTERNATIVE: Select all nodes with element in AROMATIC_ATOMS?
```

Comment:
if I recall correctly at this stage all elements are capitalized; so no it wouldn't work

Reply:
Indeed, but only because read_smiles calls `fill_valence` after.